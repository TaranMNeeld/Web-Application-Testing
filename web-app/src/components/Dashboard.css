.dash-board {
    max-width: 600px;
    max-height: 500px;
    display: flex;
    flex-wrap: wrap;
    margin: auto;
    background: red;
}

.button-div {
    max-width: 200px;
}

.btn {
    width: 90px;
    height: 90px;
    margin: 5px;
    background: green;
    color: white;
}

.display {
    min-width: 200px;
    max-height: 190px;
    margin: 5px;
}

img {
    max-width: 180px;
    max-height: 180px;
    margin: 5px;
}