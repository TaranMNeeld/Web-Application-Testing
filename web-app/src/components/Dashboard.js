import React, { useState } from 'react';
import Display from './Display';

const Dashboard = () => {
    let [ball, setBall] = useState(0);
    let [strike, setStrike] = useState(0);

    return (
        <div className='dash-board'>
            <Display ball={ball} strike={strike} />
            <button className='btn' onClick={() => {
                setBall(0);
                setStrike(0);
                console.log(`ball: ${ball}`);
                console.log(`strike: ${strike}`);
            }}>Hit</button>
            <button className='btn' name="ball" onClick={() => {
                if (ball < 4)
                    setBall(++ball);
                else
                    setBall(0);
                console.log(`strike: ${strike}`);
            }}>Ball</button>
            <button className='btn' onClick={() => {
                if (strike < 3)
                    setStrike(++strike);
                else
                    setStrike(0);
                console.log(`strike: ${strike}`);
            }}>Strike</button>
            <button className='btn' onClick={() => {
                if (strike < 2)
                    setStrike(++strike);
                console.log(`strike: ${strike}`);
            }}>Foul</button>
        </div>
    );
};

export default Dashboard;