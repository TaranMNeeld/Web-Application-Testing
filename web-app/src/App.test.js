import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Dashboard from './components/Dashboard';

let btn = document.querySelectorAll('button');

console.log(btn);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('contains home touchdown button', () => {
  const { getByTestId } = render(<App />);
  getByTestId(/home-touchdown/i);
});