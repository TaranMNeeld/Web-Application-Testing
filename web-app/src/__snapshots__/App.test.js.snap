// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dashboard component 1`] = `undefined`;

exports[`renders the display component 1`] = `undefined`;
